numpy/core/src/multiarray/ctors.c

```diff
@@ -756,14 +756,6 @@ PyArray_NewFromDescr_int(
         for (int i = 0; i < nd; i++) {
             fa->dimensions[i] = dims[i];
 
-            if (fa->dimensions[i] == 0) {
-                /*
-                 * Compare to PyArray_OverflowMultiplyList that
-                 * returns 0 in this case.
-                 */
-                continue;
-            }
-
```

Comment:
This is a subtle change in the way NumPy allocates. Until now `np.empy([2, 0, 2], np.int32) would allocate space for 4 elements. Now it will allocate space for a single `int32`.

Reply:
Am I right in thinking we had one or two PRs about this in the past?

Reply:
Ah yes, #15788 was one of them.

Reply:
OK, reverting.